Ch 01. Arrays and Strings/Chapter1.playground/Pages/1.1 Is Unique.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,30 @@
+/*:
+ 1.1 Determine if a string has all unique characters. What if you can't use additional data structures?
+ */
+extension String {
+    
+    func hasUniqueCharacters() -> Bool {
```

Comment:
As a curiosity, what would the performance difference be between your solution and the following:

```func hasUniqueCharacters() -> Bool {
        return characters.count == Set(characters).count
```}

Reply:
The running time of that implementation would always be O(n), even in cases where the set contained duplicate characters. In the implementation above, the loop stops at the first duplicate character.